### PR TITLE
Adding context title to the list view

### PR DESCRIPTION
**Pull request by jreys**
Pull Request for Issue #94 -> Add "Component: Item Type" title to views
### Testing Instructions

Go to the list view and change item type, it should change in title, was also tested in a RTL language.

### CODE REVIEW

**Comment by infograf768** [comment]
Works fine here.

The only aspect I don't like so much is Menus / Menus, but it is also the case in the Select Item Type selector.
I wonder if we should not make an exception for Menus and change to `Menu Items` and therefore get in the selector as well as in the page title `Menu / Menu Items` with the string
`COM_MENUS_SUBMENU_ITEMS="Menu Items"`

**Comment by jreys** [comment]
I think with this new title, we can replace itemType's title with Items again, so it will be:
Menus / Items, no need to hardcode for menus, just change that in the helper

**Comment by infograf768** [comment]
We would not like change what we get for example for sermonspeaker (where we get the right title)

screen shot 2016-08-18 at 12 30 08

**Comment by jreys** [comment]
the change I proposed is only regarding menus, the rest is fine

**Comment by andrepereiradasilva** [comment]
the string used is `COM_MENUS_ITEMS`

is from here: https://github.com/joomla-projects/gsoc16_improved-multi-lingual/blob/staging/administrator/components/com_associations/helpers/associations.php#L208

actually this code could be better (use plural forms) instead of `. 'S'`

**Comment by andrepereiradasilva** [comment]
@infograf768 can you check that?

**Comment by infograf768** [comment]
> actually this code could be better (use plural forms) instead of . 'S'

Not sure I get it. We are not in a case where we can use JText::plural

But we can make it a requirement (and propose another PR to core...) with a special string constant for each component concerned.
In our case (Menus) we do not have that string:
COM_MENUS_ITEMS="Menu items"

**Comment by infograf768** [comment]
I suggest to add the string above (COM_MENUS_ITEMS="Menu items") for menus in this PR.

**Comment by andrepereiradasilva** [comment]
For instance if you have an Item type called SILLY it will look for a string called SILLYS and not SILLIES as it should.

how do we solve that?

**Comment by andrepereiradasilva** [comment]
@infograf768 please chek this com_finder helper funcion: https://github.com/joomla/joomla-cms/blob/staging/administrator/components/com_finder/helpers/language.php#L28

File: administrator/language/en-GB/en-GB.com_associations.ini [hunk 1]
```diff
@@ -25,8 +25,9 @@ COM_ASSOCIATIONS_FILTER_MENUTYPE_LABEL="Menu"
 COM_ASSOCIATIONS_FILTER_SEARCH_DESC="Search an item by it's title"
 COM_ASSOCIATIONS_FILTER_SEARCH_LABEL="Search item"
 COM_ASSOCIATIONS_FILTER_SELECT_ITEM_TYPE="- Select Item Type -"
-COM_ASSOCIATIONS_HEADER_EDIT="Multilingual Associations: Associations"
-COM_ASSOCIATIONS_HEADER_SELECT_REFERENCE="Multilingual Associations: Select Reference"
+COM_ASSOCIATIONS_HEADER_EDIT="Multilingual Associations: Edit Associations (%1s =&gt; %2s)"
+COM_ASSOCIATIONS_HEADER_SELECT_FILTERS="Multilingual Associations: Select Item Type and Language"
+COM_ASSOCIATIONS_HEADER_SELECT_REFERENCE="Multilingual Associations: Select Reference (%1s =&gt; %2s)"
```
Comment (andrepereiradasilva) [inline administrator/language/en-GB/en-GB.com_associations.ini]: remove the ": Select Reference" here please

File: administrator/language/en-GB/en-GB.com_menus.ini [hunk 2]
```diff
@@ -123,6 +123,7 @@ COM_MENUS_ITEM_IS_DEFAULT="Is default"
 COM_MENUS_ITEM_MODULE_ASSIGNMENT="Module Assignment"
 COM_MENUS_ITEM_REQUIRED="Required"
 COM_MENUS_ITEM_ROOT="Menu Item Root"
+COM_MENUS_ITEMS="Menu items"
```
Comment (andrepereiradasilva) [inline administrator/language/en-GB/en-GB.com_menus.ini]: this is a core patch shouldn't be here

File: administrator/components/com_associations/views/associations/view.html.php [hunk 3]
```diff
@@ -163,9 +163,17 @@ public function display($tpl = null)
 	 */
 	protected function addToolbar()
 	{
-		$user  = JFactory::getUser();
+		$user    = JFactory::getUser();
```
Comment (andrepereiradasilva) [inline administrator/components/com_associations/views/associations/view.html.php]: why the spaces?

Comment (jreys) [inline administrator/components/com_associations/views/associations/view.html.php]: didn't remove them, was aligning with old variable, will remove them

File: administrator/language/en-GB/en-GB.com_associations.ini [hunk 4]
```diff
@@ -25,8 +25,9 @@ COM_ASSOCIATIONS_FILTER_MENUTYPE_LABEL="Menu"
...
 COM_ASSOCIATIONS_FILTER_SEARCH_LABEL="Search item"
 COM_ASSOCIATIONS_FILTER_SELECT_ITEM_TYPE="- Select Item Type -"
-COM_ASSOCIATIONS_HEADER_EDIT="Multilingual Associations: Associations"
-COM_ASSOCIATIONS_HEADER_SELECT_REFERENCE="Multilingual Associations: Select Reference"
+COM_ASSOCIATIONS_HEADER_EDIT="Multilingual Associations: Edit Associations (%1s =&gt; %2s)"
+COM_ASSOCIATIONS_HEADER_SELECT_FILTERS="Multilingual Associations: Select Item Type and Language"
+COM_ASSOCIATIONS_HEADER_SELECT_REFERENCE="Multilingual Associations (%1s =&gt; %2s)"
 COM_ASSOCIATIONS_HEADING_ASSOCIATION="Associations"
 COM_ASSOCIATIONS_HEADING_MENUTYPE="Menu"
 COM_ASSOCIATIONS_HEADING_MENUTYPE_ASC="Menu ascending"
```
Comment (andrepereiradasilva) [inline administrator/language/en-GB/en-GB.com_associations.ini:30]: why are you changing the file permissions to 755 ?

File: administrator/language/en-GB/en-GB.com_associations.ini [hunk 5]
```diff
@@ -25,8 +25,9 @@ COM_ASSOCIATIONS_FILTER_MENUTYPE_LABEL="Menu"
 COM_ASSOCIATIONS_FILTER_SEARCH_DESC="Search an item by it's title"
 COM_ASSOCIATIONS_FILTER_SEARCH_LABEL="Search item"
 COM_ASSOCIATIONS_FILTER_SELECT_ITEM_TYPE="- Select Item Type -"
-COM_ASSOCIATIONS_HEADER_EDIT="Multilingual Associations: Associations"
-COM_ASSOCIATIONS_HEADER_SELECT_REFERENCE="Multilingual Associations: Select Reference"
+COM_ASSOCIATIONS_HEADER_EDIT="Multilingual Associations: Edit Associations (%1s =&gt; %2s)"
+COM_ASSOCIATIONS_HEADER_SELECT_FILTERS="Multilingual Associations: Select Item Type and Language"
+COM_ASSOCIATIONS_HEADER_SELECT_REFERENCE="Multilingual Associations (%1s =&gt; %2s)"
```
Comment (andrepereiradasilva) [inline administrator/language/en-GB/en-GB.com_associations.ini]: maybe better?:

```
COM_ASSOCIATIONS_TITLE_EDIT="Multilingual Associations: Edit Associations (%1s &gt; %2s)"
COM_ASSOCIATIONS_TITLE_LIST="Multilingual Associations (%1s &gt; %2s)"
COM_ASSOCIATIONS_TITLE_LIST_SELECT="Multilingual Associations: Select Item Type and Language"
```

File: administrator/language/en-GB/en-GB.com_associations.ini (hunk 4 above)
Comment (jreys) [inline administrator/language/en-GB/en-GB.com_associations.ini:30]: Didn't know git stored permissions

File: administrator/language/en-GB/en-GB.com_associations.ini [hunk 6]
```diff
@@ -45,4 +43,7 @@ COM_ASSOCIATIONS_SELECT_MENU="- Select Menu -"
 COM_ASSOCIATIONS_SELECT_TARGET="Select Target"
 COM_ASSOCIATIONS_SELECT_TARGET_LANGUAGE="- Select Target Language -"
 COM_ASSOCIATIONS_TITLE="Associations"
+COM_ASSOCIATIONS_TITLE_EDIT="Multilingual Associations: Edit Associations (%1s &gt; %2s)"
+COM_ASSOCIATIONS_TITLE_LIST="Multilingual Associations (%1s &gt; %2s)"
+COM_ASSOCIATIONS_TITLE_LIST_SELECT="Multilingual Associations: Select Item Type and Language"
```
Comment (andrepereiradasilva) [inline administrator/language/en-GB/en-GB.com_associations.ini]: if you change here ... don't you need to change elsewhere ... ?

**Comment by andrepereiradasilva** [comment]
merged

**Pull request merged**